crates/bevy_ecs/src/world/error.rs

```diff
@@ -1,12 +1,62 @@
-//! Contains error types returned by bevy's schedule.
+//! Contains error types returned by methods on [`World`].
+//!
+//! [`World`]: crate::world::World
+
+use std::any::TypeId;
 
 use thiserror::Error;
 
-use crate::schedule::InternedScheduleLabel;
+use crate::{component::ComponentId, prelude::*, schedule::InternedScheduleLabel};
 
 /// The error type returned by [`World::try_run_schedule`] if the provided schedule does not exist.
 ///
 /// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
 #[derive(Error, Debug)]
 #[error("The schedule with the label {0:?} was not found.")]
 pub struct TryRunScheduleError(pub InternedScheduleLabel);
+
+/// The error type returned by [`World::get_reflect`] and [`World::get_reflect_mut`].
+///
+/// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
+#[cfg(feature = "bevy_reflect")]
+#[derive(Error, Debug)]
+pub enum GetComponentReflectError {
+    /// There is no [`ComponentId`] corresponding to the given [`TypeId`].
+    ///
+    /// Did you call [`App::register_type()`]?
+    ///
+    /// [`App::register_type()`]: ../../../bevy_app/struct.App.html#method.register_type
+    #[error(
+        "No `ComponentId` corresponding to `TypeId` {0:?} found (did you call App::register_type()?)"
```

Comment:
Nit: I think the `Debug` impl for `TypeId` includes the name `TypeId`, so you can actually remove that to avoid double-printing `TypeId`:

```
... to `TypeId` TypeId { t: (14409314272611517422, 9222282133268799408) } found ...
```

Same goes for the other error messages.

Reply:
Yep, I was thinking about that as well, but kept them in "just to be sure". But yeah, I agree that it's double-printing, so removed the additional `TypeId` now, and just leave it up to the Debug-impl :+1: